lib/datasource.js

```diff
@@ -2594,6 +2594,74 @@ DataSource.prototype.ping = function(cb) {
   return cb.promise;
 };
 
+/**
+ * Execute an arbitrary command. The commands are connector specific,
+ * please refer to the documentation of your connector for more details.
+ *
+ * @param command String|Object The command to execute, e.g. an SQL query.
+ * @param [args] Array Parameters values to set in the command.
+ * @param [options] Object Additional options, e.g. the transaction to use.
+ * @param [callback] Function The callback to call when done.
+ * @promise
+ */
+DataSource.prototype.execute = function(command, args, options, cb) {
```

Comment:
Considering that this new API will be available to LB4 only (no LB3/LB2), can we make it promise-only? By removing the callback variant, the implementation can be greatly simplified.

Of course, the API contract of connector's `execute` method remains unchanged & callback-based.